### ISSUE TITLE
Question: best way to run Odoo 12.0 on Python 3.6 or 3.7, Debian Strech aging...

### ISSUE THREAD

**Issue by rvalyi** [camptocamp/docker-odoo-project#149]
Hello,

The current Docker image for Odoo 12.0 run Python 3.5.3 on Debian 9 - Strech.

But:

- Python 3.5 has reached end of life (https://www.python.org/downloads/release/python-353/)
- Security updates have been discontinued on Debian Stretch as of July 6th, 2020 (https://www.debian.org/releases/stretch/)
May be not so much of an issue considering it still has LTS support (https://wiki.debian.org/LTS) and it is containerized so few things are exposed.

And also, I have a requirement to run Odoo 12 on Python 3.6 or 3.7 to use Data Classes (available on 3.6 via backport).

Considering all this what would you recommend?

1. Run Odoo 12.0 on the Odoo 13.0 Docker image? (does it work well?)
2. Upgrade the current Odoo 12.0 Image to Debian Buster?
3. Other like compile Python 3.7 on Debian Strech (this is easy but this would probably mean fork the image instead of reusing it)

**Comment by rvalyi** [comment]
related https://github.com/camptocamp/docker-odoo-project/issues/148

**Comment by bealdav** [comment]
I'm interested for answer on this question. cc @yvaucher @guewen @simahawk thanks

**Comment by yvaucher** [comment]
Usually we get inline with the supported version by Odoo. Though I understand your concerns.



>   1. Run Odoo 12.0 on the Odoo 13.0 Docker image? (does it work well?)

I have no experience with this, you might need to watch carefully the python dependencies though.
Those are most likely not the same and could lead to some incompatibilities:
https://github.com/camptocamp/docker-odoo-project/blob/master/13.0/base_requirements.txt

>   2. Upgrade the current Odoo 12.0 Image to Debian Buster?

This would need some testing. Could be the best option. Maybe creating a separate build? we could create a new docker tag for strech in case it has still some use.

>   3. Other like compile Python 3.7 on Debian Strech (this is easy but this would probably mean fork the image instead of reusing it)

It could be worth trying to base it on an already prepared image like python:3.7-stretch.


To sum up we probably want to create an extra image either by using stretch with a more up to date python version or by using buster, or both? We certainly don't want thousand of different images and limit to the most important ones.
But we could sub-class the images with something like docker-odoo-project:12.0-buster-4.0.0, that way it's an opt-in to switch to buster and won't bust your stability. What do you think?

**Comment by rvalyi** [comment]
Hello @yvaucher glad you are willing to move on that one. I vote for 2: upgrade do Debian Buster. I believe Odoo 12.0 is already quite tested on it as it is still pretty propular, just like Debian Buster.

I understand you prefer keeping a Debian Strech image too even if I believe we won't need it at Akretion because our v12 projects are still somewhat active and Debian Strech + EOL Python 3.5 will slowly turn into a maintenance nightmare for those who won't move on and we will still need to install Odoo or OCA modules on it.

**Comment by bealdav** [comment]
Thanks @yvaucher  for your reply.
What are the step to have less maintenance ?
1/ create a subimage on buster: with an additionnal log message to invite to test/use the next one
2/ after X month drop the old one 

If @camptocamp-business-robot accept it, can @akretion make a PR ?
cc @sebastienbeau @hparfr

Thanks

**Comment by yvaucher** [comment]
@bealdav I made a simple stupid try here:

https://github.com/camptocamp/docker-odoo-project/pull/150
But it needs some tweaking the build is failing. (more info in the PR)

**Comment by bealdav** [comment]
Thanks a lot @yvaucher
